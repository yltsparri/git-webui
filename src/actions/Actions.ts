export default class {
  public static readonly UPDATE_COMMIT_VIEW_DATA = 'UPDATE_COMMIT_VIEW_DATA';
  public static readonly TOGGLE_SHOW_FULL_FILE = 'TOGGLE_SHOW_FULL_FILE';
  public static readonly TOGGLE_IGNORE_WHITESPACE = 'TOGGLE_IGNORE_WHITESPACE';
  public static readonly UPDATE_BASEDATA = 'UPDATE_BASEDATA';
  public static readonly ADD_MESSAGE = 'ADD_MESSAGE';
  public static readonly CLOSE_MESSAGE = 'CLOSE_MESSAGE';
  public static readonly SET_LOCAL_BRANCHES = 'SET_LOCAL_BRANCHES';
  public static readonly SET_REMOTE_BRANCHES = 'SET_REMOTE_BRANCHES';
  public static readonly SET_TAGS_BRANCHES = 'SET_TAGS_BRANCHES';
  public static readonly SET_COMMITS = 'SET_COMMITS';
};
