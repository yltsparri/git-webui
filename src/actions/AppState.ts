/*
 * Copyright 2017 Ãœlo Parri
 * Copyright 2015 Eric ALBER
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { CommitInfo } from "../actions/git/CommitInfo";
import {
  CommitDiff,
  Commits,
  CommitTree,
  CommitViewMode,
  DiffOptions
} from "./Commit";
import { Diff } from "./git/Diff";
import { FileInfo } from "./git/FileInfo";
import { GitBranch } from "./git/GitBranch";

export {
  Diff,
  FileInfo,
  CommitInfo,
  GitBranch,
  CommitDiff,
  Commits,
  CommitTree,
  CommitViewMode,
  DiffOptions
};

export enum AppMode {
  Workspace = 0,
  RemoteAccess = 1,
  History = 2,
  Explore = 3
}

export class NavigationType {
  public static readonly Workspace = "workspace";
  public static readonly RemoteAccess = "remote-access";
  public static readonly LocalBranches = "local-branches";
  public static readonly RemoteBranches = "remote-branches";
  public static readonly Tags = "tags";
}

export interface AppData {
  dirName: string;
  viewOnly: boolean;
  mode: AppMode;
  messages: string;
  showAll: string;
}

export interface Offset {
  top: number;
  left: number;
}

export interface NavigationNode {
  id: string;
  text: string;
  parentId: string;
  children: string[];
  props: { [key: string]: any };
}

export interface Navigation {
  rootNodes: string[];
  nodes: { [id: string]: NavigationNode };
  showAll?: string;
  selected: string;
}

export interface AppState {
  appData: AppData;
  navigation: Navigation;
  commitDiff: CommitDiff;
  commits: Commits;
  commitTree: CommitTree;
  diffOptions: DiffOptions;
  offsets: { [key: string]: Offset };
}
