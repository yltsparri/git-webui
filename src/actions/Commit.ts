import { FileDiff } from './git/Diff';
import FileInfo from './git/FileInfo';
import CommitInfo from './git/CommitInfo';

export enum CommitViewMode {
  Diff = 0,
  SidebySideDiff = 1,
  Tree = 2
}

export interface DiffOptions {
  ignoreWhitespace: boolean;
  context: number;
  fullFile: boolean;
  options?: Array<string>;
}

export interface CommitTree {
  path: Array<FileInfo>;
  files: Array<FileInfo>;
}

export interface CommitDiff {
  fileDiffs: Array<FileDiff>;
  selectedFile: number;
  headerLines: Array<string>;
  useSplitDiff: boolean;
  removedLinesDiff: FileDiff;
  addedLinesDiff: FileDiff;
}

export interface Path {
  key: string;
  commands: Array<{ type: string, x: number, y: number }>;
}

export interface Graph {
   paths: Array<Path>;
   circles: Array<Circle>;
}

export interface Circle {
  cx: number;
  cy: number;
  r: number;
  key: string;
}

export interface Commits {
  commits: Array<CommitInfo>;
  selectedCommit: string;
  viewMode: CommitViewMode;
  graph: Graph;
}

export default Commits;
