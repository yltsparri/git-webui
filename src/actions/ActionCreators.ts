/*
 * Copyright 2017 Ãœlo Parri
 * Copyright 2015 Eric ALBER
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { CommitInfo } from './git/CommitInfo';
import GitBranch from './git/GitBranch';
import GitBrancesResponse from './git/GitBrancesResponse';
import BranchStatus from './git/BranchStatus';
import FileInfo from './git/FileInfo';
import Git from './git/Git';
import { AppMode, AppState, DiffViewMode } from './AppState';

const git = new Git();
const loadDiff = (commit: CommitInfo) => {
  return (dispatch, getState: () => AppState) => {
    const state = getState() as AppState;
    const { diffContext, ignoreWhitespace, gitDiffOpts, gitFile } = state.historyViewOptions;
    return git.getDiff(commit.hash, diffContext, ignoreWhitespace, gitDiffOpts, gitFile)
      .then(diff => {
        dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { diff: diff.data } });
        if (diff.message) {
          dispatch({ type: 'ADD_MESSAGE', message: diff.message });
        }
      });
  };
};

export function loadTree(parent: string) {
  return (dispatch) => {
    return git.listFiles(parent)
      .then(response => {
        dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { files: response.data } });
        if (response.message) {
          dispatch({ type: 'ADD_MESSAGE', message: response.message });
        }
      });
  };
};

export function loadNode(node: FileInfo) {
  return (dispatch, getState: () => AppState) => {
    let state = getState() as AppState;
    let index = state.historyViewOptions.path.findIndex(item => item.objectId === node.objectId);
    if (index >= 0) {
      if (index < state.historyViewOptions.path.length - 1) {
        dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { path: state.historyViewOptions.path.slice(0, index + 1), files: [] } });
      }
    }
    else {
      dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { path: state.historyViewOptions.path.concat(node), files: [] } });
    }
    if (node.type === 'tree') {
      return git.listFiles(node.objectId)
        .then(response => {
          state = getState() as AppState;
          dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { files: response.data } });
          if (response.message) {
            dispatch({ type: 'ADD_MESSAGE', message: response.message });
          }
        });
    }
  };
};

export function commitSelected(commit: CommitInfo) {
  return (dispatch, getState: () => AppState) => {
    let state = getState();
    let root: FileInfo = {
      name: state.baseData.dirName,
      size: NaN,
      objectId: commit.hash,
      isSymbolicLink: false,
      mode: 0,
      type: 'tree',
      parent: null
    };
    dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { commitHash: commit.hash, path: [root], files: [] } });
    state = getState();
    if (state.historyViewOptions.diffViewMode === DiffViewMode.Diff) {
      dispatch(loadDiff(commit));
    }
    else if (state.historyViewOptions.diffViewMode === DiffViewMode.Tree) {
      dispatch(loadNode(root));
    }
  };
}

export function selectDiffViewMode(mode: DiffViewMode) {
  return (dispatch, getState: () => AppState) => {
    dispatch({ type: 'UPDATE_COMMIT_VIEW_DATA', data: { diffViewMode: mode } });
    let state = getState() as AppState;
    let opts = state.historyViewOptions;
    if (mode === DiffViewMode.Tree && opts.path.length < 2 && !opts.files.length) {
      let root: FileInfo;
      if (opts.path.length) {
        root = opts.path[opts.path.length - 1];
      }
      else {
        root = {
          name: state.baseData.dirName,
          size: NaN,
          objectId: opts.commitHash,
          isSymbolicLink: false,
          mode: 0,
          type: 'tree',
          parent: null
        };
      }
      dispatch(loadNode(root));
    }
  };
}

export function itemSelected(b) {
  return (dispatch) => {
    dispatch({ type: 'UPDATE_BASEDATA', data: b });
    if (b.mode === AppMode.LocalBranches || b.mode === AppMode.RemoteBranches || b.mode === AppMode.Tags) {
      dispatch(dispatch => git.getCommits(1000, b.selectedItem)
        .then((commits) => {
          if (commits.returnCode === 0) {
            dispatch({ type: 'SET_COMMITS', commits: commits.data });
            if (commits.data && commits.data.length) {
              dispatch(commitSelected(commits.data[0]));
            }
          }
          if (commits.message) {
            dispatch({ type: 'ADD_MESSAGE', message: commits.message });
          }
        }));
    }
  };
}

export function initState() {
  return (dispatch) => {
    return git.getDirName()
      .then(dirName => {
        if (dirName.returnCode === 0) {
          dispatch({
            type: 'UPDATE_BASEDATA',
            data: {
              dirName: dirName.data
            }
          });
          if (dirName.message) {
            dispatch({ type: 'ADD_MESSAGE', message: dirName.message });
          }
          dispatch((dispatch) => git.getViewOnly()
            .then(viewOnly => {
              dispatch({
                type: 'UPDATE_BASEDATA',
                data: {
                  viewOnly: viewOnly.data === "1"
                }
              });
              if (viewOnly.message) {
                dispatch({ type: 'ADD_MESSAGE', message: viewOnly.message });
              }
            }));

          dispatch((dispatch) => git.getLocalBranches()
            .then((branches: GitBrancesResponse) => {
              dispatch({
                type: 'SET_LOCAL_BRANCHES',
                data: branches.data
              });
              if (branches.message) {
                dispatch({ type: 'ADD_MESSAGE', message: branches.message });
              }
              return branches;
            })
            .then(branches => {
              if (branches.data) {
                let active: GitBranch = branches.data.find(b => b.status === BranchStatus.Current);
                if (active) {
                  dispatch(itemSelected({ selectedItem: active.name, mode: AppMode.LocalBranches }));
                }
                if (branches.message) {
                  dispatch({ type: 'ADD_MESSAGE', message: branches.message });
                }
              }
            }));

          dispatch((dispatch) => git.getRemoteBranches()
            .then((branches: GitBrancesResponse) => {
              dispatch({
                type: 'SET_REMOTE_BRANCHES',
                data: branches.data
              });
              if (branches.message) {
                dispatch({ type: 'ADD_MESSAGE', message: branches.message });
              }
            }));

          dispatch((dispatch) => git.getTags()
            .then((tags: GitBrancesResponse) => {
              dispatch({
                type: 'SET_TAGS_BRANCHES',
                data: tags.data
              });
              if (tags.message) {
                dispatch({ type: 'ADD_MESSAGE', message: tags.message });
              }
            }));
        }
      });
  };
}
