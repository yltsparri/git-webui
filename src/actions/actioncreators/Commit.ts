/*
 * Copyright 2017 Ãœlo Parri
 * Copyright 2015 Eric ALBER
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Action } from "redux";
import { ThunkDispatch } from "redux-thunk";
import Actions from "../Actions";
import { AppState, CommitViewMode } from "../AppState";
import { CommitInfo } from "../git/CommitInfo";
import { FileInfo } from "../git/FileInfo";
import { loadDiff } from "./Diff";
import { selectNode } from "./Tree";

export function commitSelected(commit: CommitInfo) {
  return (
    dispatch: ThunkDispatch<{}, {}, Action>,
    getState: () => AppState
  ) => {
    let state = getState();
    const root: FileInfo = {
      name: state.appData.dirName,
      size: NaN,
      objectId: commit.hash,
      isSymbolicLink: false,
      mode: 0,
      type: "tree",
      parent: undefined
    };
    dispatch({ type: Actions.SELECT_COMMIT, selectedCommit: commit.hash });

    dispatch({
      type: Actions.UPDATE_COMMIT_VIEW_DATA,
      data: { commitHash: commit.hash, path: [root], files: [] }
    });
    state = getState();
    if (state.commits.viewMode === CommitViewMode.Diff) {
      dispatch(loadDiff(commit.hash));
    } else if (state.commits.viewMode === CommitViewMode.Tree) {
      dispatch(selectNode(root));
    }
  };
}
