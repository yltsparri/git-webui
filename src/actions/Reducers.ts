/*
 * Copyright 2017 Ãœlo Parri
 * Copyright 2015 Eric ALBER
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import assign from 'lodash/object/assign';
import GitBranch from './git/GitBranch';
import { CommitInfo } from '../actions/git/CommitInfo';
import { HistoryViewOptions, BaseData, AppMode, DiffViewMode } from '../actions/AppState';
import Actions from './Actions';

export function historyViewOptionsReducer(state: HistoryViewOptions, action): HistoryViewOptions {
  switch (action.type) {
    case Actions.UPDATE_COMMIT_VIEW_DATA:
      return assign({}, state, action.data);
    case Actions.TOGGLE_SHOW_FULL_FILE:
      return assign({}, state, { fullFileDiff: !state.fullFileDiff });
    case Actions.TOGGLE_IGNORE_WHITESPACE:
      return assign({}, state, { ignoreWhitespace: !state.ignoreWhitespace });
  }
  return state || {
    commitHash: '',
    diff: null,
    diffContext: 3,
    fullFileDiff: false,
    ignoreWhitespace: false,
    diffViewMode: DiffViewMode.Diff,
    path: [],
    files: []
  };
};

export function baseDataReducer(state: BaseData, action): BaseData {
  switch (action.type) {
    case Actions.UPDATE_BASEDATA:
      return assign({}, state, action.data);
    case Actions.ADD_MESSAGE:
      return assign({}, state, { messages: (state.messages ? state.messages + '\n' : '') + action.message });
    case Actions.CLOSE_MESSAGE:
      return assign({}, state, { messages: null });
  }
  return state || {
    dirName: '',
    viewOnly: true,
    selectedItem: null,
    mode: AppMode.LocalBranches,
    messages: null
  };
};

export function getBranchDataReducer(key: string) {
  return (state: Array<GitBranch>, action): Array<GitBranch> => {
    switch (action.type) {
      case key:
        return action.data || [];
      default:
        return state || [];
    }
  };
};

export function commitsReducer(state: Array<CommitInfo>, action): Array<CommitInfo> {
  switch (action.type) {
    case 'SET_COMMITS':
      return action.commits as Array<CommitInfo> || [];
    default:
      return state || [];
  }
};
