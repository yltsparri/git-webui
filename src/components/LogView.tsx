/*
 * Copyright 2017 Ãœlo Parri
 * Copyright 2015 Eric ALBER
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from "react";
import { Graph } from "../actions/Commit";
import { CommitInfo, RefType } from "../actions/git/CommitInfo";
import { COLORS } from "./Colors";

interface LogViewProps {
  commits: CommitInfo[];
  active?: string;
  graph: Graph;
  onCommitClicked: (commit: CommitInfo) => void;
}

const classNames = {
  [RefType.Head]: "label label-warning ref",
  [RefType.Unspecified]: "label label-warning ref",
  [RefType.Remote]: "label label-danger ref",
  [RefType.Tag]: "label label-info ref"
};

const lineHeight = 54;
export default class LogView extends React.PureComponent<LogViewProps> {
  public render() {
    const graph = this.updateGraph(0);
    const left = Math.max(3, graph.maxLeft) + 1;
    const xOffset = left * 12 + "px";
    return (
      <div id="log-view">
        <div>
          {graph.svg}
          {this.props.commits.map((commit, index) => {
            const refs = commit.refs!.map(tag => {
              return (
                <span
                  key={RefType[tag.type] + "_" + tag.text}
                  className={classNames[tag.type]}
                >
                  {tag.text}
                </span>
              );
            });
            const className =
              this.props.active === commit.hash
                ? "log-entry list-group-item active"
                : "log-entry list-group-item";
            return (
              <span
                data-index={index.toString()}
                onClick={this.handleCommitClick}
                key={commit.hash}
                className={className}
                style={{ paddingLeft: xOffset, height: lineHeight + "px" }}
              >
                <header>
                  <h6>
                    <a target="_blank" href={"mailto:" + commit.author.email}>
                      {commit.author.name}
                    </a>
                  </h6>
                  {refs}
                  <span className="log-entry-date">
                    {commit.author.date.toLocaleString()}
                    &nbsp;
                  </span>
                  <span className="badge">{commit.hash.substring(0, 7)}</span>
                </header>
                <p className="list-group-item-text">{commit.message}</p>
              </span>
            );
          })}
        </div>
      </div>
    );
  }

  public updateGraph = (startAt: number) => {
    const graph = this.props.graph;
    const commits = this.props.commits;
    const xOffset = 12;
    const yOffset = (startAt + 0.5) * lineHeight;
    let maxLeft = 0;
    // Draw the graph
    const circles = graph.circles.map(entry => {
      maxLeft = Math.max(maxLeft, entry.cx);
      return (
        <circle
          cx={entry.cx * xOffset}
          cy={entry.cy * lineHeight + yOffset}
          r={entry.r}
          key={entry.key}
        />
      );
    });
    const lines = graph.paths.map((stream, index) => {
      const d = stream.commands
        .map(cmd => cmd.type + " " + cmd.x * xOffset + " " + cmd.y * lineHeight)
        .join(" ");
      const colorIndex = index < COLORS.length ? index : index % COLORS.length;
      return (
        <path style={{ stroke: COLORS[colorIndex] }} key={stream.key} d={d} />
      );
    });
    return {
      svg: (
        <svg height={lineHeight * commits.length}>
          {lines}
          {circles}
        </svg>
      ),
      maxLeft
    };
  }

  private handleCommitClick = (event: React.MouseEvent<HTMLElement>) => {
    const target = event.currentTarget as HTMLElement;
    if (target.dataset.index) {
      const index = parseInt(target.dataset.index!, 10);
      if (Number.isInteger(index)) {
        this.props.onCommitClicked(this.props.commits[index]);
      }
    }
  }
}
