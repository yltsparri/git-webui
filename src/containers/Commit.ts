import { connect } from "react-redux";
import { AnyAction, Dispatch } from "redux";
import DiffActions from "../actions/actioncreators/Diff";
import NavigationActions from "../actions/actioncreators/Navigation";
import TreeActions from "../actions/actioncreators/Tree";
import { AppMode, AppState } from "../actions/AppState";
import FileInfo from "../actions/git/FileInfo";
import CommitView, {
  CommitViewActionProps,
  CommitViewStateProps
} from "../components/CommitView";

const mapStateToProps = (state: AppState): CommitViewStateProps => {
  return {
    diff: state.commitDiff.fileDiffs,
    headerLines: state.commitDiff.headerLines,
    ignoreWhitespace: state.diffOptions.ignoreWhitespace,
    diffContext: state.diffOptions.context,
    fullFileDiff: state.diffOptions.fullFile,
    diffViewMode: state.commits.viewMode,
    path: state.commitTree.path,
    files: state.commitTree.files
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    toggleIgnoreWhiteSpace: () => {
      dispatch((DiffActions.toggleIgnoreWhiteSpace() as {}) as AnyAction);
    },
    updateDiffContext: (diffContext: number) => {
      dispatch((DiffActions.setDiffContext(diffContext) as {}) as AnyAction);
    },
    onNodeSelected: (node: FileInfo) => {
      dispatch((TreeActions.selectNode(node) as {}) as AnyAction);
    },
    toggleShowFullFile: () => {
      dispatch((DiffActions.toggleShowFullFile() as {}) as AnyAction);
    },
    onExloreClicked: () => {
      dispatch(NavigationActions.changeAppMode(AppMode.Explore));
    }
  };
};

export default connect<CommitViewStateProps, CommitViewActionProps, any>(
  mapStateToProps,
  mapDispatchToProps
)(CommitView);
