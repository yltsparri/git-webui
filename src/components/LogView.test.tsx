import * as React from "react";
import { create } from "react-test-renderer";
import { CommitInfo } from "src/actions/AppState";
import { Circle, Graph, Path } from "src/actions/Commit";
import LogView from "./LogView";

it("matches snapshot", () => {
  const graph: Graph = {
    paths: new Array<Path>(),
    circles: new Array<Circle>()
  };

  const onClick = (_: CommitInfo) => {
    return;
  };
  const component = create(
    <LogView
      commits={new Array<CommitInfo>()}
      active={""}
      graph={graph}
      onCommitClicked={onClick}
    />
  );
  expect(component).toMatchSnapshot();
});
