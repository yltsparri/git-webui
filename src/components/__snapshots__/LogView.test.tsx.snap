// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="log-view"
>
  <div>
    <svg
      height={0}
    />
  </div>
</div>
`;
